packages/ui/src/redux/slices/settings.ts

```diff
@@ -0,0 +1,29 @@
+import { createSlice, Draft, PayloadAction } from "@reduxjs/toolkit";
+
+export enum ThemeMode {
+  Dark,
+  Light,
+  System,
```

Comment:
We should define a string value for each enum value here, else it will be default to `0, 1, 2` in redux:
```suggestion
  Dark = 'dark',
  Light = 'light',
  System = 'system',
```
We later can return `ThemeMode.Dark` or `ThemeMode.Light` directly in the `ThemeProvider` instead of checking it via switch case.